### PR TITLE
[SPARK-19389][ML][PYTHON][DOC] Minor doc fixes for ML Python Params and LinearSVC

### PR DESCRIPTION
**Pull request by jkbradley**
## What changes were proposed in this pull request?

* Removed Since tags in Python Params since they are inherited by other classes
* Fixed doc links for LinearSVC

## How was this patch tested?

* doc tests
* generating docs locally and checking manually

### CODE REVIEW

**Comment by jkbradley** [comment]
@wangmiao1981  Would you mind checking this?  It has small fixes I noticed when reviewing your PR for Python LinearSVC.

File: mllib/src/main/scala/org/apache/spark/ml/classification/LinearSVC.scala [hunk 1]
```diff
@@ -47,7 +47,7 @@ private[classification] trait LinearSVCParams extends ClassifierParams with HasR
 /**
  * :: Experimental ::
  *
- * Linear SVM Classifier (https://en.wikipedia.org/wiki/Support_vector_machine#Linear_SVM)
+ * [[https://en.wikipedia.org/wiki/Support_vector_machine#Linear_SVM  Linear SVM Classifier]]
```
Comment (HyukjinKwon) [inline mllib/src/main/scala/org/apache/spark/ml/classification/LinearSVC.scala]: Would you mind changing it to

```
 * <a href="https://en.wikipedia.org/wiki/Support_vector_machine#Linear_SVM">
 * Linear SVM Classifier</a>
```

? It seems scala link annotation breaks Javadoc8 as below:

```
[error] .../spark/mllib/target/java/org/apache/spark/ml/classification/LinearSVC.java:5: error: unexpected text
[error]  * {@link https://en.wikipedia.org/wiki/Support_vector_machine#Linear_SVM  Linear SVM Classifier}
```

I tested this via `./build/sbt unidoc` with Java 8.

(Currently, it seems there are already few errors. Let me fix them with other warnings in near future).

**Comment by wangmiao1981** [comment]
@jkbradley Sorry for replying late! I was out for Chinese new year this weekend. I will check the issue mentioned by HyukjinKwon otherwise, it looks good to me.

File: mllib/src/main/scala/org/apache/spark/ml/classification/LinearSVC.scala (hunk 1 above)
Comment (wangmiao1981) [inline mllib/src/main/scala/org/apache/spark/ml/classification/LinearSVC.scala]: I checked several other scala files like, GBTRegressor.scala, DecisonTreeRegressor.scala etc. URLs are in the format of <a href>...</a>.

Comment (jkbradley) [inline mllib/src/main/scala/org/apache/spark/ml/classification/LinearSVC.scala]: Thanks!  will do

**Comment by jkbradley** [comment]
I delayed too!  I just pushed a fix.  I couldn't test it since it looks like the Java 8 doc gen has already been broken again.  (Thanks a lot for the efforts to fix it!  Btw, are you pinging the people who break it whenever you see issues?)

**Comment by HyukjinKwon** [comment]
Ah, I fixed the others in https://github.com/apache/spark/pull/16741 Yes, I understand it is super easy to break. So I am trying to inform committers or frequent contributors purely as a kind comment so that we could avoid and fix them by ourself (not at all blaming purpose).

**Comment by HyukjinKwon** [comment]
(I just rebased it based on this PR and built the javadoc8 for sure. I believe it should emit an error if this PR introduce the break but it seems not. So, LGTM for doc changes.)

**Comment by jkbradley** [comment]
OK thanks a lot @HyukjinKwon and @wangmiao1981 !
Merging with master

**Pull request closed without merging**